cypress/integration/emoji/recently_used_emoji_spec.js

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [number] indicates a test step (e.g. 1. Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+/* eslint max-nested-callbacks: ["error", 3] */
+
+import {getRandomInt} from '../../utils';
+
+describe('Recent Emoji', () => {
+    before(() => {
+        // 1. Login and go to /
+        cy.login('user-1');
+        cy.visit('/');
+    });
+
+    it('M14014 Recently used emojis are shown 1st', async () => {
+        const maxRandom = 400;
+
+        // 2. Get random emoji index
+        const firstEmoji = getRandomInt(maxRandom);
```

Comment:
While I like the randomness of `firstEmoji` and `secondEmoji`, this introduces flaky tests whenever:
- `firstEmoji` and `secondEmoji` are equal (rare but can happen) or the `secondEmoji` got a value of 1 (it will hit the recent emoji).  As a result, no recent emoji will be added on second usage.
- on local dev where you ran multiple tests of this spec, recent emojis will accumulate up to 27 emojis.  If it happened that the `firstEmoji` and/or `secondEmoji` is less than or equal to number of recent emojis, then again no recent emoji will be added.

Not great solution but just meet the test requirement, I'd suggest to have fix values like `firstEmoji = 300` and `secondEmoji = 200`.

Good to have but not necessary: clear storage before testing so that there's no recent emoji before this test.

Reply:
noted